pkg/kubelet/volumes.go

```diff
@@ -244,123 +63,17 @@ func (kl *Kubelet) getPodVolumes(podUID types.UID) ([]*volumeTuple, error) {
 		// but skipping dirs means no cleanup for healthy volumes. switching to a no-exit api solves this problem
 		volumeNameDirs, volumeNameDirsStat, err := util.ReadDirNoExit(volumeKindPath)
 		if err != nil {
-			return []*volumeTuple{}, fmt.Errorf("could not read directory %s: %v", volumeKindPath, err)
+			return false, fmt.Errorf("could not read directory %s: %v", volumeKindPath, err)
 		}
 		for i, volumeNameDir := range volumeNameDirs {
 			if volumeNameDir != nil {
-				volumes = append(volumes, &volumeTuple{Kind: volumeKind, Name: volumeNameDir.Name()})
+				volumeDirsExist = true
 			} else {
 				glog.Errorf("Could not read directory %s: %v", podVolDir, volumeNameDirsStat[i])
```

Comment:
nit: It's unclear from the description of this function whether an error should be returned here or not...